common/config/azure-pipelines/templates/publish.yaml

```diff
@@ -18,37 +18,66 @@ steps:
      import sys, os, glob, re, subprocess
      import shutil
 
-     def determineDistTag(currVersion):
+     def determineDistTag(currentVer, latestVer, previousVer):
        ## The master branch is the only one that should get the 'nightly' release tag
        mainBranch = "master"
 
-       # Branch that is the current set of releases, will be given the 'latest' dist tag
-       currentReleaseBranch = "release/2."
-
-       # Branches that used to hold the current releases but will no longer now not be given the 'latest' dist tag
-       # Note: This should probably be an array in the future.
-       oldReleaseBranch = "release/1."
-
        distTag = None
        if len(sys.argv) == 4:
          branchName = sys.argv[3]
 
-         print ("Branch name: " + branchName + " and current version " + currVersion)
+         print ("Branch name: " + branchName + "\nCurrent version: " + currentVer + "\nLatest version: " + latestVer + "\Previous version: " + previousVer)
 
          # The most common case is the tag will be a nightly tag
          if mainBranch in branchName:
            distTag = "nightly"
          elif "release/" in branchName:
            print ("On a release branch")
-           if currentReleaseBranch in branchName:
-             distTag = "latest"
-           elif oldReleaseBranch in branchName:
-             distTag = "previous"
-
-           # Override the 2 previous checks for the case where we're releasing a -dev version off a release branch.
-           # These will always be considered rc.
-           if "-" in currVersion:
+
+           currentDevVer = -1
+           latestDevVer = -1
+           previousDevVer = -1
+           if ("-" in currentVer):
+             currentVer, currentDevVer = currentVer.split("-")
+             currentDevVer = int(currentDevVer.split(".")[1])
+           if ("-" in latestVer):
+             latestVer, latestDevVer = latestVer.split("-")
+             latestDevVer = int(latestDevVer.split(".")[1])
+           if ("-" in prevousVer):
```

Comment:
this should be `previousVer` instead of `prevousVer` in a couple places within the file.

Reply:
Ah nice catch, thanks. Coding in the inline python script is a bit of a pain!